### PR TITLE
Add a tubular script for triggering a jenkins job  TE-1854

### PR DESCRIPTION
**Pull request by jzoldak**
@edx/platform-team 
Ready for review. This is the pattern for triggering a job over on Jenkins.

### CODE REVIEW

File: requirements/test_requirements.txt [hunk 1]
```diff
@@ -1,8 +1,10 @@
 ddt==1.0.1
-nose==1.3.7
+httpretty==0.8.10
```
Comment (jzoldak) [inline requirements/test_requirements.txt:2]: Reviewer note: 
httpretty must be an implicit requirement, as a test already uses it (and it was in my virtualenv). But it was not listed in the requirements file. Adding it to be explicit.

File: tubular/jenkins.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    # Some failure has happened.
+    msg = 'Call to Jenkins failed. Status: {0}, Reason: {1}.'.format(
+        response.status_code, response.reason
+    )
+    # Give helpful messages for the conditions that we know about.
+    if response.status_code == 404:
+        msg += ' Verify that you have permissions to the job and double check the spelling of its name.'
+    elif response.status_code == 405:
+        msg += ' Verify that you passed the required parameters to the job.'
```
Comment (doctoryes) [inline tubular/jenkins.py:87]: Helpful!

File: tubular/tests/test_jenkins.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    @ddt.data(
+        (None, (), None),
+        ('my cause', (), [u'cause=my+cause']),
+        (None, ((u'FOO', u'bar'),), [u'FOO=bar']),
+        (None, ((u'FOO', u'bar'), (u'BAZ', u'biz')), [u'FOO=bar', u'BAZ=biz']),
+        ('my cause', ((u'FOO', u'bar'),), [u'cause=my+cause', u'FOO=bar']),
+    )
+    @ddt.unpack
+    @httpretty.activate
+    def test_success(self, cause, param, expected_query):
```
Comment (doctoryes) [inline tubular/tests/test_jenkins.py:51]: Maybe add a `test_failure` that checks for the `BackendError`?

**Review by doctoryes — APPROVED** [review]
👍

**Pull request merged**